### PR TITLE
Add pytest-bdd

### PR DESCRIPTION
**Pull request by breathe**
Hi python review team: `@conda-forge/help-python`

Would like to get this pytest extension into conda-forge.  meta.yaml is adapted from `conda skeleton pypi` -- I added 

```
build:
    noarch: python 
```

As I believe the package does not have any native extension dependencies.

Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages
- [X] Build number is 0
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there (@breathe is me and I am willing)

### CODE REVIEW

File: recipes/pytest-bdd/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+build:
+  noarch: python
+  number: 0
+  entry_points:
+    - pytest-bdd = pytest_bdd.scripts:main
+  script: "{{ PYTHON }} setup.py install --single-version-externally-managed --record=record.txt"
+
+requirements:
+  build:
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]: This should be host requirements:
```suggestion
  host:
```

File: recipes/pytest-bdd/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+{% set hash_value = "17dfc2b65c275641a4e76fa1f913ab737604815c275dc2afe7133956c6ab6743" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]: As these are only used once:
```suggestion
  sha256: 17dfc2b65c275641a4e76fa1f913ab737604815c275dc2afe7133956c6ab6743
```

File: recipes/pytest-bdd/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+{% set file_ext = "tar.gz" %}
+{% set hash_type = "sha256" %}
+{% set hash_value = "17dfc2b65c275641a4e76fa1f913ab737604815c275dc2afe7133956c6ab6743" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]: This isn't needed
```suggestion
```

File: recipes/pytest-bdd/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]:
```suggestion
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

File: recipes/pytest-bdd/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
+{% set name = "pytest-bdd" %}
+{% set version = "3.1.0" %}
+{% set file_ext = "tar.gz" %}
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]:
```suggestion
```

File: recipes/pytest-bdd/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+{% set hash_type = "sha256" %}
+{% set hash_value = "17dfc2b65c275641a4e76fa1f913ab737604815c275dc2afe7133956c6ab6743" %}
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]:
```suggestion
```

File: recipes/pytest-bdd/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,65 @@
+{% set name = "pytest-bdd" %}
+{% set version = "3.1.0" %}
+{% set file_ext = "tar.gz" %}
+{% set hash_type = "sha256" %}
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]:
```suggestion
```

File: recipes/pytest-bdd/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  number: 0
+  entry_points:
+    - pytest-bdd = pytest_bdd.scripts:main
+  script: "{{ PYTHON }} setup.py install --single-version-externally-managed --record=record.txt"
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - glob2
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]: You probably only need most of these as `run` dependencies. You should start from `python` and `pip` and work from there.

File: recipes/pytest-bdd/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  noarch: python
+  number: 0
+  entry_points:
+    - pytest-bdd = pytest_bdd.scripts:main
+  script: "{{ PYTHON }} setup.py install --single-version-externally-managed --record=record.txt"
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]: Can you use `pip` like the example does?

File: recipes/pytest-bdd/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  requires:
+    - tox
+
+about:
+  home: https://github.com/pytest-dev/pytest-bdd
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE.txt
+  summary: BDD for pytest
+  description: ""
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]: If these don't have values they should be removed.

File: recipes/pytest-bdd/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    - py
+    - pytest >=3.0.0
+    - six >=1.9.0
+
+test:
+  imports:
+    - pytest_bdd
+  commands:
+    - pytest-bdd --help
+  requires:
```
Comment (chrisburr) [inline recipes/pytest-bdd/meta.yaml]: It looks like you don't need `tox` as you're not using it for testing?

Comment (breathe) [reply recipes/pytest-bdd/meta.yaml]: Nope don't need it

File: recipes/pytest-bdd/meta.yaml (hunk 10 above)
Comment (breathe) [reply recipes/pytest-bdd/meta.yaml]: I added values for these fields

File: recipes/pytest-bdd/meta.yaml (hunk 9 above)
Comment (breathe) [reply recipes/pytest-bdd/meta.yaml]: Yup

File: recipes/pytest-bdd/meta.yaml (hunk 8 above)
Comment (breathe) [reply recipes/pytest-bdd/meta.yaml]: Great thanks!

**Comment by breathe** [comment]
Hi thanks for the feedback!

I incorporated your suggestions in https://github.com/conda-forge/staged-recipes/commit/4c179d26489eb7ee1a6253e3682b6f875b68aa34

Apologies if I didn't use right workflow for incorporating your suggestions (haven't used github review and inline edit/commit tools much)

**Comment by chrisburr** [comment]
You can use the web interface to apply them or do it manually like you did and either option is fine from my perspective.

LGTM and welcome to conda-forge 🎉

**Comment by breathe** [comment]
Sweet -- I see the package up -- https://anaconda.org/conda-forge/pytest-bdd

But I can't seem to install it ...?  

```
conda install -c conda-forge pytest-bdd

Collecting package metadata: done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - pytest-bdd

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/free/osx-64
  - https://repo.anaconda.com/pkgs/free/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.```

**Comment by chrisburr** [comment]
According to the Gitter chat there is an issue with updating the CDN, it should be fixed soon.

**Pull request merged**